snuba/query/query.py

```diff
@@ -216,6 +218,12 @@ def get_prewhere(self) -> Sequence[Condition]:
         """
         return self.__prewhere_conditions
 
+    def get_prewhere_ast(self) -> Optional[Expression]:
```

Comment:
As long as prewhere processor is not a Clickhouse query processor, this needs to be fully supported.